xbmc/utils/LangCodeExpander.h

```diff
@@ -140,6 +141,20 @@ class CLangCodeExpander
   std::vector<std::string> GetLanguageNames(LANGFORMATS format = ISO_639_1,
                                             LANG_LIST list = LANG_LIST::DEFAULT);
 
+  /** \brief Performs various ISO 3166-1 lookups.
+  *          Input can be alpha-2, alpha-3 or name.
+  *          If the input is alpha-2 or alpha-3 then the
+  *          name will be returned unless returnCode is true,
+  *          then the opposite code will be returned.
+  *          If the input is a name, then alpha-3 is returned
+  *          unless returnCode is true, then alpha-2 is returned.
+  *   \param[in] saught The code or name being saught.
+  *   \param[out] returned The code or name being returned.
+  *   \param[in] returnCode Control the value returned.
+  *   \return true if the conversion succeeded, false otherwise.
+  */
+  bool LookupISO31661(const std::string& saught, std::string& returned, bool returnCode = false);
```

Comment:
`const std::string& saught` seem a good candicate to be optimized with `string_view`:
https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#1211-stdstring-vs-stdstring_view

Reply:
I will look into this one.

Reply:
@CastagnaIT - It would appear that `StringUtils::ToLower` (required for a case-blind search) does not appear to currently support `std::string_view` objects.  Your guidance in the matter would be greatly appreciated.

Reply:
as `StringUtils::ToLower(saught.data())`
utils methods will be improved in near future to use string_view

Reply:
@CastagnaIT - Thanks, that worked perfectly.